src/pages/create/Opportunity.tsx

```diff
@@ -0,0 +1,249 @@
+            </H2>
+          )}
+          <InputWithTitle
+            title="Next Step"
+            register={register("NextStep")}
+            error={!!errors.NextStep}
+          />
+          <DateField
+            label="Close Date"
+            onChange={(e: any) => setValue("CloseDate", e[0].toISOString())}
+            required
+            error={!!errors.CloseDate}
+          ></DateField>
+          <DropdownSelect
+            title="Stage"
+            required
+            data={
+              opportunityMetadata.data?.fields.find(
+                (e) => e.name === "StageName"
+              )?.picklistValues
+            }
+            keyName="label"
+            valueName="value"
+            value={stage}
+            error={!!errors.StageName}
+            onChange={(e) => setValue("StageName", e)}
+          />
+        </Stack>
+        <H0>Additional Information</H0>
+        <Stack vertical gap={12} style={{ width: "100%" }}>
+          <InputWithTitle
+            title="Order Number"
+            register={register("OrderNumber")}
+            error={!!errors.OrderNumber}
+          />
+          <DropdownSelect
+            title="Delivery/Installation Status"
+            data={
+              opportunityMetadata.data?.fields.find(
+                (e) => e.name === "DeliveryInstallationStatus__c"
+              )?.picklistValues
+            }
+            keyName="label"
+            valueName="value"
+            value={dlStatus}
+            error={!!errors.DeliveryInstallationStatus__c}
+            onChange={(e) => setValue("DeliveryInstallationStatus__c", e)}
+          />
+          <InputWithTitle
+            title="Tracking Number"
+            register={register("TrackingNumber__c")}
+            error={!!errors.TrackingNumber__c}
+          />
+          <InputWithTitle
+            title="Main Competitors"
+            register={register("MainCompetitors__c")}
+            error={!!errors.MainCompetitors__c}
+          />
+          <InputWithTitle
+            title="Current Generator"
+            register={register("CurrentGenerators__c")}
+            error={!!errors.CurrentGenerators__c}
+          />
+          <TextArea
+            variant="inline"
+            title="Description"
+            value={description}
+            error={!!errors.Description}
+            onChange={(e) => setValue("Description", e.target.value)}
+            placeholder="Enter text here..."
+            style={{
+              resize: "none",
+              minHeight: "5em",
+              maxHeight: "100%",
+              height: "auto",
+              width: "100%",
+              overflow: "hidden",
+            }}
+          />
+        </Stack>
+        <Stack
+          style={{ justifyContent: "space-between", width: "100%" }}
+          gap={5}
+        >
+          <Button
+            type="submit"
+            text={submitting ? "Creating..." : "Create"}
```

Comment:
In this case, we usually `disabled` button and show the `loading` icon

here example https://github.com/DeskproApps/gitlab/blob/master/src/components/IssueForm/IssueForm.tsx#L127